File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/util/ENIntervalUtil.java
package org.coralibre.android.sdk.internal.datatypes.util;

import org.coralibre.android.sdk.internal.datatypes.ENInterval;

import static org.coralibre.android.sdk.internal.EnFrameworkConstants.TEK_INTERVAL_LENGTH_SECONDS;
import static org.coralibre.android.sdk.internal.EnFrameworkConstants.TEK_ROLLING_PERIOD;

public class ENIntervalUtil {

    public static ENInterval getMidnight(final ENInterval enInterval) {
        return new ENInterval( ((long)(enInterval.get() / TEK_ROLLING_PERIOD)) * TEK_ROLLING_PERIOD);
    }

    public static long getMidnight(final long rawENNumber) {
        return ((rawENNumber/ TEK_ROLLING_PERIOD) * TEK_ROLLING_PERIOD);
    }

    public static ENInterval createFromUnixTimestamp(long unixtime) {
        return new ENInterval( unixtime / TEK_INTERVAL_LENGTH_SECONDS );
    }

    public static long intervalNumberFromUnixTimestamp(long unixtime) {
        return unixtime / TEK_INTERVAL_LENGTH_SECONDS;
    }

    public static long intervalNumberToUnixTimestamp(long interval) {
        return interval * TEK_INTERVAL_LENGTH_SECONDS;
    }

    public static ENInterval getCurrentInterval() {
        return new ENInterval(System.currentTimeMillis() / 1000L, true);
    }

}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/util/ENIntervalUtil.kt
package org.coralibre.android.sdk.internal.datatypes.util

import org.coralibre.android.sdk.internal.EnFrameworkConstants
import org.coralibre.android.sdk.internal.datatypes.ENInterval

object ENIntervalUtil {
    @JvmStatic
    fun getMidnight(enInterval: ENInterval): ENInterval {
        return ENInterval(
            (enInterval.get() / EnFrameworkConstants.TEK_ROLLING_PERIOD) * EnFrameworkConstants.TEK_ROLLING_PERIOD
        )
    }

    @JvmStatic
    fun getMidnight(rawENNumber: Long): Long {
        return rawENNumber / EnFrameworkConstants.TEK_ROLLING_PERIOD * EnFrameworkConstants.TEK_ROLLING_PERIOD
    }

    @JvmStatic
    fun createFromUnixTimestamp(unixtime: Long): ENInterval {
        return ENInterval(unixtime / EnFrameworkConstants.TEK_INTERVAL_LENGTH_SECONDS)
    }

    @JvmStatic
    fun intervalNumberFromUnixTimestamp(unixtime: Long): Long {
        return unixtime / EnFrameworkConstants.TEK_INTERVAL_LENGTH_SECONDS
    }

    @JvmStatic
    fun intervalNumberToUnixTimestamp(interval: Long): Long {
        return interval * EnFrameworkConstants.TEK_INTERVAL_LENGTH_SECONDS
    }

    @JvmStatic
    val currentInterval: ENInterval
        get() = ENInterval(System.currentTimeMillis() / 1000L, true)
}
